staff/demi/spotify-app-react/script/main_.js

```diff
@@ -0,0 +1,164 @@
+(function() {  
+  "use strict";
+  //spotifyApi.timeout = 5000;
+
+  /**
+   *  Enviamos datos a la api de sporify
+   */
+  // llamamos a la busqueda y a la api le pasaremos en los handles la query y las 2 funciones 1 de OK otra de KO
+  $('form').submit(function (e) {
+    e.preventDefault();
+    const query = $('input').val();
+    const baseUrl= "https://api.spotify.com/v1/"
+    spotifyApi.searchArtists(query, baseUrl).then(artists=>showResults(artists))
+  })
+
+  // definimos la funcion showResults
+  function showResults(artists){
+    let boxCard="";
+    const ulrImg ="";
+    $(".section-track").hide()
+    $(".section-tracks").hide()
+    
+      artists.forEach(function(item) {
+        
+        const imagesUrl = item.images.length !=0 ? item.images[0].url : '';
+
+        boxCard+="<div class='col p-2 card'>";
+        boxCard+="<div class='photobox photobox-artists photobox_type16' data-url='https://api.spotify.com/v1/artists/"+item.id+"/albums'>";
+        boxCard+="<div class='photobox__previewbox'>";
+        boxCard+="<img class='img-fluid photobox__preview card-img' src="+imagesUrl+">";
+        boxCard+="<span class='photobox__label'>"+item.name+"</span>";
+        boxCard+="</div>";
+        boxCard+="</div>";
+        boxCard+="</div>";
+        
+      });
+
+    $("section.section-artists .row").html(boxCard);
+    $("section.section-artists .row").show();
+    $(".section-artists").show()
+
+  }
+
+  // llamamos al siguiente pantel donde le pasaremos 
+  // los mismos datos de token time out base url... 
+  // pero las funciones seran ditintas y la query
+
+  /**
+   *  Enviamos datos a la api de sporify
+   */
+  $("body").on("click", ".photobox-artists", (function(e) { 
+    const query = $(this).attr("data-url")
+    const baseUrl= "https://api.spotify.com/v1/"
+    spotifyApi.retrieveAlbums(query,baseUrl).then(albums=>showAlbums(albums))
+  }))
+  
+  //definimos la funcion showAlbums que nos servira para enseñar los albums
+  function showAlbums(albums) {
+
+    $(".section-artists").fadeIn(1000).hide();
+    $(".section-album").fadeIn(1000).show();

```

Comment:
SHOULD $ elements be defined at the upper scope level and reused here instead of call $(...) each time. that is:

// definitions at the upper scope level 
const $artistsSection = $('.section-artists')
const $albumsSection = $('.section-albums')

// usage at the lower scope levels
$artistsSection.fadeIn(...
$albumsSection.fadeIn(...